### PR TITLE
[Feature] mParticle Migration Update (SDK v7+)

### PR DESCRIPTION
**Pull request by yoyoyoseob**
Updates the initialization method to support mParticle version 7+

Based on a call with a mParticle rep -- opted to leave the **MParticleOptions** out of Simcoe to make it explicit that the consumer is responsible for handling errors and such

"The SDK will not retry API calls automatically, but provides callback APIs such that you can do so according to your business logic."

Included are some links for reference:

1. [Initialization](https://docs.mparticle.com/developers/sdk/ios/initialize-the-sdk)
2. [User Identity](https://docs.mparticle.com/developers/sdk/ios/identity)
3. [Error Handling](https://docs.mparticle.com/developers/sdk/ios/identity#error-handling)

### CODE REVIEW

**Comment by yoyoyoseob** [comment]
@MichaelCampbell 

Was wondering if you knew who I would contact for the `Travis Errors` 

Seems to be caused by differing cocoapod versions?

File: Simcoe.podspec [hunk 1]
```diff
@@ -34,7 +34,7 @@ Pod::Spec.new do |s|
   # Each subspec represents an analytics library implemented using Simcoe.
 
   adobe		  =   { :name => "Adobe",           :dependency => "AdobeMobileSDK", :version => '~> 4.13' }
```
Comment (MichaelCampbell) [inline Simcoe.podspec:36]: seems like the spacing is messed up here

Comment (yoyoyoseob) [reply Simcoe.podspec:36]: What should it be? I assumed it was intentional to vertically align the values

Comment (MichaelCampbell) [reply Simcoe.podspec:36]: i'm referring to this spacing:
```
adobe		  =
```
possibly it's just displaying oddly on GH?

Comment (yoyoyoseob) [reply Simcoe.podspec:36]: Oh no it is like that in the podspec, seems intentional 
screen shot 2018-08-07 at 2 23 09 pm

File: .travis.yml [hunk 2]
```diff
@@ -3,17 +3,16 @@
 # # * https://github.com/supermarin/xcpretty#usage
 #
 
-language: objective-c
-osx_image: xcode9
-cache: cocoapods
+language: swift
+osx_image: xcode9.3
+# cache: cocoapods
```
Comment (MichaelCampbell) [inline .travis.yml]: seems like we might be able to cache again, since the image was the issue

**Review by MichaelCampbell — APPROVED** [review]
LGTM

File: .travis.yml (hunk 2 above)
Comment (yoyoyoseob) [reply .travis.yml]: Yup sound good!

**Pull request merged**